### DISCUSSION TITLE
Is there a stable forgot password link for clients to use?

### DISCUSSION THREAD

**Discussion by alechenninger** [keycloak/keycloak#11865] [Q&A]
A client can link directly to login-actions/reset-credentials endpoint to start the forgot password flow. Is this endpoint considered a stable API for clients or is it considered implementation detail / might change in a Keycloak release?

**Comment by abstractj** [comment]
Hi @Alechenninger, I'm not sure if it answers your question. But we don't have any plans to redesign the API currently, so `login-actions/reset-credentials` should be considered safe/stable.

**Reply by alechenninger** [reply]
I guess the question is more if that is intended to be linked to, directly, by clients? I think the answer is "no" actually–for example if you try to use the back to login flow link, it errors. This makes sense to me because the login flow is within a protocol and a client, however just going to the reset credentials page specifies neither a protocol nor a client. On the other hand, if all you want to do is reset your password, you don't need those other things. So perhaps there is a feature somewhere here to enable direct link to reset password (without the extra links that cause errors when not used within a proper auth flow)?

**Reply by stianst** [reply]
> Hi @Alechenninger, I'm not sure if it answers your question. But we don't have any plans to redesign the API currently, so `login-actions/reset-credentials` should be considered safe/stable.

Wrong. It's not a documented or public endpoint, it's an internal endpoint, and as such can change without any warning.

**Comment by kiriakos** [comment]
This is a great question!

Actually, I would like to specifically extend this to  the OIDC variant of reset-credentials:
`/auth/realms/{REALM}/protocol/openid-connect/forgot-credentials`

Main reason being that this endpoint seems to honor query params like `redirect_uri` better than `login-actions/reset-credentials`.

**Comment by stianst** [comment]
Login actions is very much and internal endpoint and can't be considered a public endpoint.

**Reply by kiriakos** [reply]
What's the official documentation on public endpoints? 
Presumably everything that `/auth/realms/{REALM}/.well-known/openid-configuration` links ist a public endpoint for OIDC?
What would this mean for `/auth/realms/{REALM}/protocol/openid-connect/forgot-credentials` which is not explicitly returned in openid-configuration but falls unter the same base path as the other resources returned by it.
